### ISSUE TITLE
Issue with AuthService property not being passed into Callback component

### ISSUE THREAD

**Issue by johnhunsley** [auth0-samples/auth0-vue-samples#31]
After login it successfully calls the callback URL - http://localhost:8080/callback this is routed to the Callback.vue component but the auth instance of AuthService is undefined - 

vue.esm.js?efeb:476[Vue warn]: Error in data(): "TypeError: Cannot read property 'handleAuthentication' of undefined"

found in

---> <Callback> at src\components\Callback.vue
       <App> at src\App.vue
         <Root>

**Comment by kbzone** [comment]
After some debugging of this issue, I see that the problem is in here:

`
<router-view 
        :auth="auth" 
        :authenticated="authenticated">
</router-view>
`

(https://github.com/auth0-samples/auth0-vue-samples/blob/master/01-Login/src/App.vue#L32)

The router-view dont delegate the props, because as @johnhunsley said this error appears on callback.vue, but in the Home.vue too when we init the app, when you click on the "Log In" anchor.

**Comment by kbzone** [comment]
i get an answer from @jmangelo with this fix:
https://community.auth0.com/answers/10202/view

tl;dr: update your package.json with this exact version and re-install dependencies:
"vue": "2.3.4"
"vue-router": "2.7.0"
"vue-template-compiler": "2.3.4"

**Comment by johnhunsley** [comment]
Works. I also had to upgrade webpack -   "webpack": "^3.7.1"

**Comment by MrClan** [comment]
No need to downgrade any packages. Simply making a minor change inside Callback.vue, by removing props, and accessing AuthService through normal import fixes the issue. Replace the script block with this: 

```Javascript
  import AuthService from './../auth/AuthService.js'
  const auth = new AuthService()

  export default {
    name: 'callback',
    data () {
      auth.handleAuthentication()
      return {}
    }
  }
```
